app/controllers/users_controller.rb

```diff
@@ -32,18 +32,14 @@ def create
 
   def confirm
     user = User.find_by_id_and_confirmation_token(params[:id], params[:token])
+    logout if logged_in? && user
     if user.nil?
       redirect_to login_url, alert: 'Bad url'
     elsif !user.confirmed_at.blank?
```

Comment:
ok, this code was older than confirmed? method, my bad I didn't pay attention to that line